src/NUnitEngine/nunit.engine.api/ITestAgentInfo.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+using System.Collections.Generic;
+
+namespace NUnit.Engine.Services
+{
+    /// <summary>
+    /// Objects implementing ITestAgentInfo by classes that can return information about
+    /// available agents. The interface is used by runners in order to provide information
+    /// to the user or to allow selecting agents.
+    /// </summary>
+    public interface ITestAgentInfo
```

Comment:
I'm not a huge fan of TestAgentInfo not implementing ITestAgentInfo.  I feel like `ITestAgentInfoProvider` or something is more clear.